lib/node_modules/@stdlib/blas/ext/base/wasm/dnansumkbn2/test/test.module.ndarray.js

```diff
@@ -0,0 +1,233 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/* eslint-disable node/no-sync */
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var Memory = require( '@stdlib/wasm/memory' );
+var Float64Array = require( '@stdlib/array/float64' );
+var Module = require( './../lib' ).Module;
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof Module, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'a module instance has a `ndarray` method which has an arity of 4', function test( t ) {
+	var mem;
+	var mod;
+
+	mem = new Memory({
+		'initial': 0
+	});
+	mod = new Module( mem );
+	t.strictEqual( mod.ndarray.length, 4, 'returns expected value' );
+	t.end();
+});
+
+tape( 'a module instance has a `ndarray` method that the sum of absolute values of all strided array elements', function test( t ) {
+	var mem;
+	var mod;
+	var xp;
+	var y;
+
+	mem = new Memory({
+		'initial': 1
+	});
+	mod = new Module( mem );
+	mod.initializeSync();
+
+	xp = 0;
+
+	mod.write( xp, new Float64Array( [ 1.0, -2.0, NaN, 5.0, 0.0, 3.0 ] ) );
+
+	y = mod.ndarray( 6, xp, 1, 0 );
+	t.strictEqual( y, 7.0, 'returns expected value' );
+
+	t.end();
+});
+
```

Comment:
See `blas/ext/base/dnansumkbn2`. A new set of tests has been added for signed zero.

Reply:
Note that you may need to also regenerate the Wasm binary, as the upstream implementation has changed.